src/admanager.js

```diff
@@ -132,11 +132,10 @@ exports = new (Class(Emitter, function () { // jshint ignore:line
     is_dismissed = false;
   };
 
-  this.showAd = function () {
+  this.showAd = function (spent_time, ad_freq_secs) {
 
     // assigning cached ad to chosen
-    if (chosen) {
-
+    if (chosen && spent_time > ad_freq_secs) {
```

Comment:
Just moving the check here, doesn't look like it is separating a lot of logic out